src/Adapter/StockManager.php

```diff
@@ -102,6 +103,11 @@ public function isAsmGloballyActivated()
     public function updatePhysicalProductQuantity($shopId, $errorState, $cancellationState, $idProduct = null, $idOrder = null)
     {
         $this->updateReservedProductQuantity($shopId, $errorState, $cancellationState, $idProduct, $idOrder);
+        $whereShopIdCond = $shopId;
+
+        if(Shop::isFeatureActive() && Shop::getContextShopGroup()->share_stock){
```

Comment:
You can use `ShopAdapter` class here and do `if ($shop_group->share_stock) {` take a look at the code in above methods. Also, no need to check if multi-store is enabled there, I think that will be handled anyway.